src/snowflake/connector/network.py

```diff
@@ -958,6 +958,21 @@ def _request_exec_wrapper(
             retry_ctx.increment_retry(reason)
             if retry_ctx.timeout is not None:
                 retry_ctx.timeout -= sleeping_time
+            if "Connection aborted" in repr(e) and "ECONNRESET" in repr(e):
```

Comment:
I wonder if there is an error code that we could use to determine whether we want to recreate the connection?

Reply:
Here's the hierarchy of the error raised by the requests library on my mac:

<img width="672" alt="Screenshot 2023-07-13 at 2 27 51 PM" src="https://github.com/snowflakedb/snowflake-connector-python/assets/104799010/42e53b8a-4c9a-4f77-bea7-0b3536316b30">

honestly I choose to compare the error message because of the simplicity of implementation, the error hierarchy of requests is unclear to me, so if I want to get the error code, I have to try catch and check attribute, and fall back to check error message if there's no such attribute.